src/regex.c

```diff
@@ -19,69 +19,72 @@
 #include "datastructure.h"
 #include <regex.h>
 
-static int num_regex;
-static regex_t *regex = NULL;
-static bool *regexconfigured = NULL;
-static char **regexbuffer = NULL;
+static int num_regex[2] = { 0 };
+static regex_t *regex[2] = { NULL };
+static bool *regexconfigured[2] = { NULL };
+static char **regexbuffer[2] = { NULL };
 
-static void log_regex_error(const char *where, const int errcode, const int index)
+static const char regextype[2][10] = { "whitelist", "blacklist" };
+
+static void log_regex_error(const char *where, const int errcode, const int index, const unsigned char regexid)
 {
 	// Regex failed for some reason (probably user syntax error)
 	// Get error string and log it
-	const size_t length = regerror(errcode, &regex[index], NULL, 0);
+	const size_t length = regerror(errcode, &regex[regexid][index], NULL, 0);
 	char *buffer = calloc(length,sizeof(char));
-	(void) regerror (errcode, &regex[index], buffer, length);
-	logg("ERROR %s regex on line %i: %s (%i)", where, index+1, buffer, errcode);
+	(void) regerror (errcode, &regex[regexid][index], buffer, length);
+	logg("ERROR %s regex %s no. %i: %s (%i)", where, regextype[regexid], index+1, buffer, errcode);
```

Comment:
Instead of `no.`, `id = ` may be clearer.

Reply:
The regex no. 0 might be not what is at table ID = 0. FTL starts to count drom zero on for the first regex it read from the table. However, there could be a regex with ID=0 which is either disabled or has been removed. Then "ID" would be misleading. We could modify the view to return both the ID and the regex string (instead of currently only the domain, `CREATE VIEW vw_regex_blacklist AS SELECT DISTINCT domain`). I don't think we could get a meaningful ID, otherwise.

The obvious alternative is to remove this as we can print the failed regex string itself.

Reply:
It would be easier for the user to debug if we print the regex instead of the number or ID.

Reply:
With the latest commit, we will print
```
[2019-07-16 12:24:47.778 4694] ERROR: compiling regex whitelist filter "?": Invalid preceding regular expression (13)
```
when the compilation fails.

However, we will have to store them permanently if we also want to be able to report failures during runtime evaluation. Currently, we store the string representation of the filters only if `DEBUG_REGEX` is enabled.

I will investigate on if it is at all possible to encounter an error during runtime if a filter was compiled successfully.

Reply:
Updated the log message to be shown as a warning rather than an error (it doesn't really affect the DNS functionality):
```
[2019-07-16 22:31:52.558 10489] Warning: Invalid regex whitelist filter "?": Invalid preceding regular expression (error code 13)
```